hw1/slavrom.js

```diff
@@ -0,0 +1,203 @@
+        };
+
+        addNumbers = (n) => {
+            const num = n.toString().split('');
+            let result = [];
+            num.forEach(function(i){
+                result.push(+i)
+            })
+            return result.reduce((sum,current) => sum + current);
+        }
+
+        sameNumbers = (n) => {
+            const num = n.toString().split('').sort();
+            for (let i = 0; i < num.length; i++) {
+                if (num[i] == num[i + 1]) {
+                    return `There is a double digit: ${num[i]}`;
+                }
+            }
+            return 'There is no double digit';
+        }
+
+        sameHalf = (n) => {
+            let number = n.toString().split('');
+            let firstHalf = +number[0] + +number[1];
+            let secondHalf = +number[2] + +number[3];
+            if (firstHalf == secondHalf) {
+                return 'Same halves !';
+            }else {
+                return 'Different halves !';
+            }
+        }
+
+        return res;
+    }else {
+        return 'You can use only four-digit number !'
+    }
+ 
+}
+
+console.log(checkNumber(5436));
+console.log(checkNumber(5456));
+console.log(checkNumber(54565));
+
+// or
+
+class Number {
+    constructor(number) {
+        this.number = number;
+    }
+
+    checkFourDiggitNumber = () => {
+        const strN = this.number.toString();
+        if (strN.length === 4) {
+            const res = {
+                sum: this.addNumbers(this.number),
+                same: this.sameNumbers(this.number),
+                firstLastSame: this.sameHalf(this.number)
+            };
+            return res
+        }
+    }
+
+    addNumbers = () => {
+        const num = this.number.toString().split('');
+        let result = [];
+        num.forEach(function(i){
+            result.push(+i)
+        })
+        return result.reduce((sum,current) => sum + current);
+    }
+
+    sameNumbers = () => {
+        const num = this.number.toString().split('').sort();
+        for (let i = 0; i < num.length; i++) {
+            if (num[i] == num[i + 1]) {
+                return `There is a double digit: ${num[i]}`;
+            }
+        }
+        return 'There is no double digit';
+    }
+
+    sameHalf = () => {
+        const number = this.number.toString().split('');
+        let firstHalf = +number[0] + +number[1];
+        let secondHalf = +number[2] + +number[3];
+        if (firstHalf == secondHalf) {
+            return 'Same halves !';
+        }else {
+            return 'Different halves !';
+        }
+    }
+}
+
+let aNumber = new Number(1234);
+
+console.log(checkFourDiggitNumber(aNumber));
+
+// 5
+
+const statement1 = [1, 2, 3] + {}; // 1,2,3[object Object]
+
+const statement2 = 1 + '3' - 2; // 11
+
+const statement3 = 1 + ('3' - 2); // 2
+
+let a = 2; // 4
+let b = a++; // 2
+let c = ++a; // 4
+
+// 6
+
+checkDataType = (i) => {
+    return Object.prototype.toString.call(i).toLowerCase();
```

Comment:
The only thing which is missing is to slice the part of string which corresponds to the data type and return it.